### PR TITLE
docs: changelog for next version

### PR DESCRIPTION
**Pull request by henryiii**
## Description

I think it's about time for another release. I'm interested in getting a few of the build system fixes out (Windows ARM support, specifiably), and it looks like there have been some other nice fixes.

Preparing the changelog for the next release. Is there any reason to make this a minor release? Everything looks pretty safe for a patch release to me. The one large change is the Eigen one, but that's a large change to a small optional portion, and backward compatible, I think.

### CODE REVIEW

**Comment by rwgk** [comment]
> The one large change is the Eigen one, but that's a large change to a small optional portion, and backward compatible, I think.

Yes, agreed. But I think it'll be best to finalize and merge #4412 before making a release, to fix the ODR issue in the test that accidentally slipped in.

**Comment by rwgk** [comment]
> Yes, agreed. But I think it'll be best to finalize and merge #4412 before making a release, to fix the ODR issue in the test that accidentally slipped in.

I just merged #4412.

File: docs/changelog.rst [hunk 1]
```diff
@@ -15,6 +15,62 @@ IN DEVELOPMENT
...
+* ``pybind11/eigen/tensor.h`` adds converters to and from ``Eigen::Tensor`` and
+  ``Eigen::TensorMap``.
+  `#4201 <https://github.com/pybind/pybind11/pull/4201>`_
+
+* Add option for enable/disable enum members in docstring.
+  `#2768 <https://github.com/pybind/pybind11/pull/2768>`_
+
+Bug fixes:
+
+
```
Comment (rwgk) [inline docs/changelog.rst:35]: Is the 2nd empty line intentional?

File: docs/changelog.rst [hunk 2]
```diff
@@ -15,6 +15,62 @@ IN DEVELOPMENT
...
+* Bug fix affecting only Python 3.6 under very specific, uncommon conditions:
+  move ``PyEval_InitThreads()`` call to the correct location.
+  `#4350 <https://github.com/pybind/pybind11/pull/4350>`_
+
+* Fixed typing of ``KeysView``, ``ValuesView`` and ``ItemsView`` in ``bind_map``.
```
Comment (rwgk) [inline docs/changelog.rst:40]: This is more a feature change than a bug fix. Gray area. But I'd list it under Changes above.

File: docs/changelog.rst [hunk 3]
```diff
@@ -15,6 +15,62 @@ IN DEVELOPMENT
...
+  `#4353 <https://github.com/pybind/pybind11/pull/4353>`_
+
+* ``PyGILState_Check()``'s  were integrated to ``pybind11::handle``
```
Comment (rwgk) [inline docs/changelog.rst:43]: This helps catching (other) bugs: this one I'd definitely move up under Changes.

File: docs/changelog.rst [hunk 4]
```diff
@@ -15,6 +15,62 @@ IN DEVELOPMENT
...
+  ``inc_ref()`` & ``dec_ref()``. The added GIL checks are guarded by
+  ``PYBIND11_ASSERT_GIL_HELD_INCREF_DECREF``, which is the default only if
+  ``NDEBUG`` is not defined.
+  `#4246 <https://github.com/pybind/pybind11/pull/4246>`_
+
+* Fix segfault bug when passing foreign native functions to functional.h.
```
Comment (rwgk) [inline docs/changelog.rst:49]: @Lalaland for comment:

IIUC strictly speaking the segfault bug is only fixed `if PYBIND11_INTERNALS_VERSION > 4`, which is not the default.

I'm not sure if there are more nuances, and what makes sense to explain here.

**Review by rwgk — APPROVED** [review]

File: docs/changelog.rst (hunk 4 above)
Comment (Lalaland) [reply docs/changelog.rst:49]: The segfault bug is partially fixed here for abi version 4. Remember, I added two unit tests for the bugfix that both failed under the old code

One of the unit tests now passes all the time. The other unit test only passes with the new abi version.

Comment (Lalaland) [reply docs/changelog.rst:49]: I would leave this entry as is, and then have a new entry "More functional.h segfault fixes" when we do an ABI upgrade

File: docs/changelog.rst (hunk 1 above)
Comment (henryiii) [reply docs/changelog.rst:43]: No, though it's a bit random currently on older entries too. When we move to markdown I'll try to use prettier.

**Review by Skylion007 — APPROVED** [review]

**Pull request merged**